pandas/core/frame.py

```diff
@@ -8016,8 +8016,8 @@ def _arith_method_with_reindex(self, right: DataFrame, op) -> DataFrame:
             right.columns, how="inner", level=None, return_indexers=True
         )
 
-        new_left = left.iloc[:, lcols]
-        new_right = right.iloc[:, rcols]
+        new_left = left if lcols is None else left.iloc[:, lcols]
```

Comment:
Can also return left if `lcols.equals(left.columns)`? Or is that when `None` is returned?

Reply:
`lcols` is a positional indexer and yes when lcols is `None` that would be when `cols.equals(left.columns)`